datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -955,6 +965,36 @@ impl PushDownFilter {
     }
 }
 
+/// Convert cross join to join by pushing down filter predicate to the join condition
```

Comment:
Looks like this function only does "Convert cross join to join"? 

"pushing down filter predicate" is done by `push_down_all_join`. Newbies might be confused by this.

Reply:
Exactly